ledger/participant-integration-api/src/main/scala/platform/store/appendonlydao/events/ContractStateEventsReader.scala

```diff
@@ -0,0 +1,200 @@
+      maybeCreateKeyValueCompression: Option[Int],
+  ): Option[Key] =
+    for {
+      createKeyValue <- maybeCreateKeyValue
+      createKeyValueCompression = Compression.Algorithm.assertLookup(
+        maybeCreateKeyValueCompression
+      )
+      keyValue = decompressAndDeserialize(createKeyValueCompression, createKeyValue)
+    } yield GlobalKey.assertBuild(templateId, keyValue.value)
+
+  private def decompressAndDeserialize(algorithm: Compression.Algorithm, value: InputStream) =
+    ValueSerializer.deserializeValue(algorithm.decompress(value))
+
+  private val createsAndArchives: EventsRange[Long] => SimpleSql[Row] =
+    (range: EventsRange[Long]) => SQL(s"""
+                                         |SELECT
+                                         |    event_kind,
+                                         |    contract_id,
+                                         |    template_id,
+                                         |    create_key_value,
+                                         |    create_key_value_compression,
+                                         |    create_argument,
+                                         |    create_argument_compression,
+                                         |    flat_event_witnesses,
+                                         |    ledger_effective_time,
+                                         |    event_sequential_id,
+                                         |    event_offset
+                                         |FROM
+                                         |    participant_events
+                                         |WHERE
+                                         |    event_sequential_id > ${range.startExclusive}
+                                         |    and event_sequential_id <= ${range.endInclusive}
+                                         |    and (event_kind = ${EventKind.Create} or event_kind = ${EventKind.ConsumingExercise})
+                                         |ORDER BY event_sequential_id ASC
+                                         |""".stripMargin)
+
+  case class CreateMissingError(field: String) extends NoStackTrace {
+    override def getMessage: String =
+      s"Create events should not be missing $field"
+  }
+
+  case class InvalidEventKind(eventKind: Int) extends NoStackTrace {
+    override def getMessage: String =
+      s"Invalid event kind: $eventKind"
+  }
+
+  private[events] case class RawContractStateEvent(
+      eventKind: Int,
+      contractId: ContractId,
+      templateId: Option[Ref.Identifier],
+      ledgerEffectiveTime: Option[Instant],
+      createKeyValue: Option[InputStream],
+      createKeyCompression: Option[Int],
+      createArgument: Option[InputStream],
+      createArgumentCompression: Option[Int],
+      flatEventWitnesses: Set[Party],
+      eventSequentialId: Long,
+      offset: Offset,
+  )
+
+  private object EventKind {
+    val Create = 10
+    val ConsumingExercise = 20
```

Comment:
Please inline the constants.

Reply:
These constants are used both in a SQL query and in pattern matching when processing rows retrieved from the database. I would change the query to use hardcoded values while keeping these constants for pattern matching. What do you think @gerolf-da ?

Reply:
That sounds good. thanks.